source/aerobulk/flux.py

```diff
@@ -45,29 +207,69 @@ def flux_noskin(
         by default 2
     niter : int, optional
         Number of iteration steps used in the algorithm,
-        by default 1
+        by default 6
 
     Returns
     -------
-    ql : array_like
+    ql : xarray.DataArray
         Latent heat flux                                     [W/m^2]
-    qh : array_like
+    qh : xarray.DataArray
         Sensible heat flux                                   [W/m^2]
-    taux : array_like
+    taux : xarray.DataArray
         zonal wind stress                                    [N/m^2]
-    tauy : array_like
+    tauy : xarray.DataArray
         meridional wind stress                                    [N/m^2]
-    evap : array_like
+    evap : xarray.DataArray
         evaporation         [mm/s] aka [kg/m^2/s] (usually <0, as ocean loses water!)
     """
     _check_algo(algo, VALID_ALGOS)
-    ql, qh, taux, tauy, evap = aero.mod_aerobulk_wrapper.aerobulk_model_noskin(
-        algo, zt, zu, sst, t_zt, hum_zt, u_zu, v_zu, slp, niter
+
+    sst, t_zt, hum_zt, u_zu, v_zu, slp = xr.broadcast(
+        sst, t_zt, hum_zt, u_zu, v_zu, slp
     )
-    return ql, qh, taux, tauy, evap
 
+    if len(sst.dims) < 3:
+        # TODO promote using expand_dims?
+        raise NotImplementedError
+    if len(sst.dims) > 4:
+        # TODO iterate over extra dims? Or reshape?
+        raise NotImplementedError
+
+    out_vars = xr.apply_ufunc(
+        noskin_np,
+        sst,
+        t_zt,
+        hum_zt,
+        u_zu,
+        v_zu,
+        slp,
+        input_core_dims=[()] * 6,
+        output_core_dims=[()] * 5,
```

Comment:
On a theoretical level the calculation is pointwise ([this test](https://github.com/xgcm/aerobulk-python/blob/5df2ff4a9db36af5da5f4da0bb161819eb4d6549/tests/test_flux_xr.py#L73-L94) confirms that). For practical reasons we definitely do not want to define time as a core dim, since it would prevent us from using `dask='parallelized'` in most cases. 
The restrictions (of passing 3d arrays) can be solved independent of the actual dimensions, so I think it is not appropriate to define core dims here.